### PR TITLE
Fixing separable metric documentation

### PR DESCRIPTION
**Pull request by dcoeurjo**
# PR Description

This PR fixes wrong complexities in higher dimension for the separable volumetric algorithms (Voronoi, distance transform..) in the documentation.

# Checklist

- [x] Unit-test of your feature with [Catch](http://dgtal.org/doc/stable/moduleCatch.html).
- [x] Doxygen documentation of the code completed (classes, methods, types, members...)
- [x] Documentation module page added or updated.
- [x] New entry in the [ChangeLog.md](https://github.com/DGtal-team/DGtal/blob/master/ChangeLog.md) added.
- [x] No warning raised in Debug ```cmake``` mode (otherwise, Travis C.I. will fail).
- [x] All continuous integration tests pass (Travis & appveyor)

### CODE REVIEW

File: ChangeLog.md [hunk 1]
```diff
@@ -60,6 +53,8 @@
     [#1330](https://github.com/DGtal-team/DGtal/pull/1330))
   - Checking boost version when including boost/common_factor_rt (David Coeurjolly,
     [#1344](https://github.com/DGtal-team/DGtal/pull/1344))
+  - Fix cmputational costs of separable metric predicates in the documentation.
```
Comment (rolanddenis) [inline ChangeLog.md]: c**o**mputational :ok_hand:

File: src/DGtal/geometry/doc/moduleMetrics.dox [hunk 2]
```diff
@@ -216,27 +216,29 @@ As discussed in @ref moduleVolumetric, both VoronoiMap and PowerMap (and their a
...
+| ExactPredicateLpSeparableMetric specialized for p=1 | @f$ O(d) @f$ | @f$ O(d log(n)) @f$ | Exact computations|
+| InexactPredicateLpSeparableMetric with p at construction | @f$ O(d) @f$ | @f$ O(d  log(n)) @f$ | Inexact computations since std::pow on double is used (supposed to be @f$ O(1) @f$)|
+| ExactPredicateLpPowerSeparableMetric<p> | @f$ O(d\cdot log(p)) @f$ | @f$ O(d log(p)log(n)) @f$ | Exact computations|
+| ExactPredicateLpPowerSeparableMetric specialized for p=2 | @f$ O(d) @f$ | @f$ O(d) @f$ | Exact computations|
 | SeparableMetricAdapter of a metric aMetric  | @f$ O(m) @f$ | @f$ O(mlog(n)) @f$ | relies on aMetric(p,q) computations |
 | experimental::ChamferNorm2D | @f$ O(log(p)) @f$ | @f$ O(log^2(n)) @f$ | @cite dcoeurjo_metric14|
 
 
-Following this table, VoronoiMap, DistanceTransformation, PowerMap, ReverseDistanceTransformation have the following computational cost:
+Following this table, VoronoiMap and DistanceTransformatio  have the following computational cost:
```
Comment (rolanddenis) [inline src/DGtal/geometry/doc/moduleMetrics.dox]: DistanceTransformatio**n**

File: ChangeLog.md (hunk 1 above)
Comment (dcoeurjo) [reply ChangeLog.md]: thx :)

File: src/DGtal/geometry/doc/moduleMetrics.dox (hunk 2 above)
Comment (dcoeurjo) [reply src/DGtal/geometry/doc/moduleMetrics.dox]: thx

**Comment by dcoeurjo** [comment]
thanks @rolanddenis should be ok now

**Comment by dcoeurjo** [comment]
You were right, there is an issue with the dim vs domain size. I’ll fix it

**Comment by dcoeurjo** [comment]
@rolanddenis should be ok now

File: src/DGtal/geometry/doc/moduleMetrics.dox [hunk 3]
```diff
@@ -216,27 +216,29 @@ As discussed in @ref moduleVolumetric, both VoronoiMap and PowerMap (and their a
...
 |--------------------------------------------------|---|---|------|
-| ExactPredicateLpSeparableMetric<p>               | @f$ O(log(p)) @f$ | @f$ O(log(p)log(n)) @f$ | Exact computations|
-| ExactPredicateLpSeparableMetric specialized for p=2 | @f$ O(1) @f$ | @f$ O(1) @f$ | Exact computations|
-| ExactPredicateLpSeparableMetric specialized for p=1 | @f$ O(1) @f$ | @f$ O(log(n)) @f$ | Exact computations|
-| InexactPredicateLpSeparableMetric with p at construction | @f$ O(1) @f$ | @f$ O(log(n)) @f$ | Inexact computations since std::pow on double is used (supposed to be @f$ O(1) @f$)|
-| ExactPredicateLpPowerSeparableMetric<p> | @f$ O(log(p)) @f$ | @f$ O(log(p)log(n)) @f$ | Exact computations|
-| ExactPredicateLpPowerSeparableMetric specialized for p=2 | @f$ O(1) @f$ | @f$ O(1) @f$ | Exact computations|
-| SeparableMetricAdapter of a metric aMetric  | @f$ O(m) @f$ | @f$ O(mlog(n)) @f$ | relies on aMetric(p,q) computations |
-| experimental::ChamferNorm2D | @f$ O(log(p)) @f$ | @f$ O(log^2(n)) @f$ | @cite dcoeurjo_metric14|
+| ExactPredicateLpSeparableMetric<p>               | @f$ O(log(p)) @f$ | @f$ O(log(p)log(N)) @f$ | Exact computations|
```
Comment (JacquesOlivierLachaud) [inline src/DGtal/geometry/doc/moduleMetrics.dox]: Use \log instead of log in math formula.

File: src/DGtal/geometry/doc/moduleMetrics.dox [hunk 4]
```diff
@@ -216,27 +216,29 @@ As discussed in @ref moduleVolumetric, both VoronoiMap and PowerMap (and their a
...
+| ExactPredicateLpSeparableMetric specialized for p=2 | @f$ O(d) @f$ | @f$ O(d) @f$ | Exact computations|
+| ExactPredicateLpSeparableMetric specialized for p=1 | @f$ O(d) @f$ | @f$ O(d\cdot log(N)) @f$ | Exact computations (H could be implemented in @f$ O(d) @f$)|
```
Comment (JacquesOlivierLachaud) [inline src/DGtal/geometry/doc/moduleMetrics.dox]: Usually \cdot is a scalar product for vector. I would just write O(d \log(N) ). The \log will make the formula nicer.

File: src/DGtal/geometry/doc/moduleMetrics.dox [hunk 5]
```diff
@@ -216,27 +216,29 @@ As discussed in @ref moduleVolumetric, both VoronoiMap and PowerMap (and their a
...
+| InexactPredicateLpSeparableMetric with p at construction | @f$ O(d) @f$ | @f$ O(d  \cdot log(N)) @f$ | Inexact computations since std::pow on double is used (supposed to be @f$ O(1) @f$)|
```
Comment (JacquesOlivierLachaud) [inline src/DGtal/geometry/doc/moduleMetrics.dox]: Same here and below

File: src/DGtal/geometry/doc/moduleMetrics.dox [hunk 6]
```diff
@@ -216,27 +216,29 @@ As discussed in @ref moduleVolumetric, both VoronoiMap and PowerMap (and their a
...
+| ExactPredicateLpPowerSeparableMetric specialized for p=2 | @f$ O(d) @f$ | @f$ O(d) @f$ | Exact computations|
+| SeparableMetricAdapter of a metric aMetric  | @f$ O(m) @f$ | @f$ O(m  \cdot log(N)) @f$ | relies on aMetric(p,q) computations in @f$ O(m) @f$ (the metric must satisfy some constraines, see @cite dcoeurjo_metric14)|
+| experimental::ChamferNorm2D | @f$ O(log(p)) @f$ | @f$ O(log^2(N)) @f$ | @cite dcoeurjo_metric14|
 
 
-Following this table, VoronoiMap, DistanceTransformation, PowerMap, ReverseDistanceTransformation have the following computational cost:
+Following this table, VoronoiMap and DistanceTransformation  have the following computational cost:
 
- \f[  O(d\cdot n^d\cdot (C+H))  \f]
+ \f[  O(d\cdot N^d\cdot (C+H))  \f]
```
Comment (JacquesOlivierLachaud) [inline src/DGtal/geometry/doc/moduleMetrics.dox]: also here

File: src/DGtal/geometry/doc/moduleMetrics.dox (hunk 4 above)
Comment (dcoeurjo) [reply src/DGtal/geometry/doc/moduleMetrics.dox]: \cdot is also just product on scalars when you want to separate the terms. With the product of several logs, it helps the readability I think

Comment (JacquesOlivierLachaud) [reply src/DGtal/geometry/doc/moduleMetrics.dox]: As you wish !

**Comment by copyme** [comment]
A typo: consturction -> construction 

Suggestions:
exactly the same -> the same (no need to over stress, IMHO)
processes -> methods (maybe)

**Comment by copyme** [comment]
In fact, the separable (remove in face ?)

 compute the DT given -> compute the DT, given (without the comma it does not make sense)

Beside its -> Besides its
labelling -> labeling
classic tools -> standard tools

Take it easy,
Kacper

**Comment by dcoeurjo** [comment]
Typo fixed.
@Kacper, in which files are the typos you mention ?  I didn't find them...

**Comment by dcoeurjo** [comment]
Thx for the comments

**Comment by dcoeurjo** [comment]
should be ok now. Let's wait for travis
Thanks for the feedbacks

File: src/DGtal/geometry/doc/moduleMetrics.dox [hunk 7]
```diff
@@ -25,218 +24,116 @@ namespace concepts{
...
+CPowerSeparableMetric [label="CPowerSeparableMetric" URL="@ref CPowerSeparableMetric"];
+
+CSeparableMetric -> CMetricSpace;
+CSeparableMetric -> CMetricSpace;
+CDigitalSeparableMetric -> CDigitalMetricSpace;
+CDigitalSeparableMetric -> CSeparableMetric;
+CPowerSeparableMetric -> CPowerMetric;
+
+label="Main metric concepts";
+}
```
Comment (rolanddenis) [inline src/DGtal/geometry/doc/moduleMetrics.dox]: Why did you unindent these lines ? Same question about the modification below

**Comment by copyme** [comment]
@dcoeurjo moduleVolumetric.dox If I remember correctly. Sorry I should have written this in a more organized way.

File: src/DGtal/geometry/doc/moduleMetrics.dox (hunk 7 above)
Comment (dcoeurjo) [reply src/DGtal/geometry/doc/moduleMetrics.dox]: I didn’t in this PR

**Comment by dcoeurjo** [comment]
@kacper i see. It was unclear for me since these typos do not come from my PR. I’ll correct them anyway ;) thx

File: src/DGtal/geometry/doc/moduleMetrics.dox (hunk 7 above)
Comment (dcoeurjo) [reply src/DGtal/geometry/doc/moduleMetrics.dox]: My bad.. some weird side effects of my editor. Let me fix it

**Comment by dcoeurjo** [comment]
thx @copyme, fixed

**Comment by dcoeurjo** [comment]
should be good now. thanks folks

**Comment by rolanddenis** [comment]
I have made some minor fixes: modification in tabulations, space dimension in 2 formula and a typo

**Comment by rolanddenis** [comment]
BTW shouldn't we use `n` for the space dimension instead of `d` that is already used for the distance map ? :grimacing:

File: src/DGtal/geometry/doc/moduleMetrics.dox [hunk 8]
```diff
@@ -15,71 +15,60 @@
...
 
- - Chamfer norms with axis symmetric unit ball (see experimental::ChamferNorm2D in dimension 2).
+- Chamfer norms with axis symmetric unit ball (see experimental::ChamferNorm2D in dimension 2).
 
- - DigitalMetricAdapter which adapts any metric (model of
-   CMetricSpace) to digital space using the ceil function. This induces
-   a digital metric (ceil function preserves the
-   triangular inequality).
- 
+- DigitalMetricAdapter which adapts any metric (model of    CMetricSpace) to digital space using the ceil function. This induces    a digital metric (ceil function preserves the    triangular inequality).
```
Comment (rolanddenis) [inline src/DGtal/geometry/doc/moduleMetrics.dox]: Is that kind of modification voluntary or also due to the editor ? :grin:

Comment (dcoeurjo) [reply src/DGtal/geometry/doc/moduleMetrics.dox]: apparently

File: src/DGtal/geometry/doc/moduleMetrics.dox [hunk 9]
```diff
@@ -129,114 +118,79 @@ reverse distance reconstruction and medial axis extraction.  For @f$
 l_2@f$ metric, the power distance between a point @f$ x@f$ and a ball
 @f$(y,r)@f$ with centger @f$y@f$ and radius @f$r@f$, is defined by
 
-@f[ pow(x,y) = \| x - y\|_2^2  - r @f]
 
-Power function are no more positive but the geometry of the distance
-(convex unit balls,...) are relevant in volumetric analysis.
+@note Models of CMetricSpace concept also implements a  @e rawDistance(x,y) method. This method returns a value of type @e RawValue which may be considered as an exact representation of the distance between x and y. For example, if we consider a @f$ l_p@f$ metric, the distance between x and y can be represented as the sum of power p terms (the distance without the order p square root). Hence, if RawValue is an @e exact @e type (e.g. model of CInteger with enough capacity to represent power p integers), rawDistance method can be used to perform exact metric comparison.  To sum up for any instance @e metric of the class ExactLpSeparableMetric for instance:
```
[the remark this answers is not in the document]
Comment (dcoeurjo) [reply src/DGtal/geometry/doc/moduleMetrics.dox]: it is.. thx

**Comment by dcoeurjo** [comment]
for `d`, I know but the differences are petty obvious from the context

**Comment by dcoeurjo** [comment]
I'll change it if you prefer...

**Comment by copyme** [comment]
@dcoeurjo since you touched the main page in this PR you may like to fix these bugs:

a better -> better (no articles with uncountable nouns) 
shared to -> shared with
 but show already ->  but shows already 
have a look to -> have a look at
PhD -> Ph.D. (seems to be more correct)
Furthermore -> Furthermore, (missing comma)
so as to -> to
whose main -> whose primary
of effective demonstration -> of useful demonstration

Sorry for such a mess with these suggestions.

**Comment by dcoeurjo** [comment]
thanks @rolanddenis for the commit (and  @copyme for the typos)
Everything looks good now.

**Pull request merged**